controllers/extendeddaemonsetreplicaset/strategy/canary.go

```diff
@@ -164,20 +163,27 @@ func manageCanaryPodRestarts(pods []*v1.Pod, params *Parameters, result *Result)
 
 		newRestartTime      time.Time
 		restartingPodStatus string
+
+		cannotStart          bool
+		cannotStartPodStatus string
 	)
 
 	// Note that we still need to evaluate restarts regardless of the enabled autoPause or autoFail
 	// since we maintain the restarting condition that can be checked by canary.noRestartsDuration
 	for _, pod := range pods {
 		restartCount, highRestartReason := podUtils.HighestRestartCount(pod)
-		if restartCount == 0 {
-			continue
+		if restartCount != 0 {
+			restartTime, recentRestartReason := podUtils.MostRecentRestart(pod)
+			if restartTime.After(newRestartTime) {
+				newRestartTime = restartTime
+				restartingPodStatus = fmt.Sprintf("Pod %s restarting with reason: %s", pod.ObjectMeta.Name, string(recentRestartReason))
+			}
 		}
 
-		restartTime, recentRestartReason := podUtils.MostRecentRestart(pod)
-		if restartTime.After(newRestartTime) {
-			newRestartTime = restartTime
-			restartingPodStatus = fmt.Sprintf("Pod %s restarting with reason: %s", pod.ObjectMeta.Name, string(recentRestartReason))
+		var cannotStartReason v1alpha1.ExtendedDaemonSetStatusReason
+		cannotStart, cannotStartReason = podUtils.CannotStart(pod)
+		if cannotStart {
```

Comment:
could you add a counter about how many pods are note able to start? and add this counter to the description?

Reply:
We could do that but I think we should refrain from converting EDS status into a fully fledged debugging tool. Rationale for the currently implemented behavior is that we are considering it to be abnormal enough to pause the canary on first failure.